internal/datacoord/index_meta.go

```diff
@@ -421,6 +422,46 @@ func (m *indexMeta) GetSegmentIndexStateOnField(collID, segmentID, fieldID Uniqu
 	return state
 }
 
+func (m *indexMeta) GetIndexedSegments(collectionID int64, fieldIDs []UniqueID) []int64 {
+	m.RLock()
+	defer m.RUnlock()
+
+	fieldIndexes, ok := m.indexes[collectionID]
+	if !ok {
+		return nil
+	}
+
+	fieldIDSet := typeutil.NewUniqueSet(fieldIDs...)
+
+	checkSegmentState := func(indexes map[int64]*model.SegmentIndex) bool {
+		indexedFields := 0
+		for indexID, index := range fieldIndexes {
+			if !fieldIDSet.Contain(index.FieldID) {
+				continue
+			}
+
+			if index.IsDeleted {
+				return false
```

Comment:
should continue.
example:
CreateIndex A,
DropIndex A,
CreateIndex B.
we only need to pay attention to the status of index B.